lib/channel.js

```diff
@@ -0,0 +1,85 @@
+/* ################################## */
+/* ##### Channel Registration  ##### */
+/* ################################## */
+var fs = require("fs");
+
+var config = JSON.parse(fs.readFileSync("config/default.json"));
+var apiConfig = config.openhimApi;
+
+exports.registerChannel = function(channelConfig, callback){
+
+  var needle = require('needle');
+  var crypto = require('crypto');
+
+  // used to bypass self signed certificates
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+
+  // define login credentails for authorization
+  var username = apiConfig.username;
+  var password = apiConfig.password;
+  var apiURL = apiConfig.protocol+'://'+apiConfig.host+':'+apiConfig.port;
+
+  // authenticate the username
+  needle.get(apiURL+'/authenticate/'+username, function(err, resp, body) {
+
+    console.log('Attempting to create/update channel');
+
+    // print error if exist
+    if (err){
+      console.log(err)
+      return;
+    }
+
+    // if user isnt found - console log response body
+    if ( resp.statusCode !== 200 ){
+      console.log(resp.body);
+      return;
+    }
+
+    // create passhash
+    var shasum = crypto.createHash('sha512');
+    shasum.update( body.salt + password );
+    var passhash = shasum.digest('hex');
+
+    // create token
+    shasum = crypto.createHash('sha512');
+    shasum.update( passhash + body.salt + body.ts );
+    var token = shasum.digest('hex');
+
+    // define request headers with auth credentails
+    var options = {
+      json: true,
+      headers: { 'auth-username': username,
+                  'auth-ts': body.ts,
+                  'auth-salt': body.salt,
+                  'auth-token': token }
+    }
```

Comment:
Perhaps we should move to using the module that I referenced on Jira. I see you only saw that after you had done it this way, sorry about that! I think it will be better to use the module as then we just have to maintain the code in one place.